core/src/chunk.rs

```diff
@@ -59,6 +60,77 @@ pub struct Chunk {
     /// Whether this chunk has been modified since the most recent
     /// call to `check_modified`().
     modified: bool,
+
+    heightmaps: [HeightMap; CHUNK_WIDTH * CHUNK_WIDTH],
+}
+
+#[derive(Clone, Copy, Default)]
+pub struct HeightMap {
+    motion_blocking: u8,
+    motion_blocking_no_leaves: u8,
+    ocean_floor: u8,
+    ocean_floor_wg: u8,
+    world_surface: u8,
+    world_surface_wg: u8,
+}
+
+impl HeightMap {
```

Comment:
Because `HeightMap` is Copy and just plain data, I think the idiomatic thing to do in rust is to make the fields public and remove all getters and setters.  Mutability is already controlled in Rust and if the getters and setters aren't upholding invariants then they don't really serve any purpose at all in Rust.

Reply:
The doc comments on these functions are great though, those should be moved to the struct fields! (if the no-op getters/setters are removed)

Reply:
You should not be able to modify world geneated heights.

Reply:
However maybe we want to be able to modify the generated height at some point, and it should be up to the user if she/he has the need?

Reply:
After some thought, I realized that you should not be able to obtain a mutable reference of a Heightmap and therefore all fields should be public.